CMakeLists.txt

```diff
@@ -128,50 +136,104 @@ else()
     set(LIBNGRAPH_SO "libngraph.so")
 endif(APPLE)
 
+# Build options
+option(USE_PRE_BUILT_NGRAPH "Use pre-built nGraph located in NGRAPH_ARTIFACTS_DIR" FALSE)
+option(NGRAPH_ARTIFACTS_DIR "Where would nGraph be installed after build" FALSE)
+option(UNIT_TEST_ENABLE "Control the building of unit tests" FALSE)
+option(UNIT_TEST_USE_TF_PREBUILT "Use pre-built libtensorflow_cc.so" FALSE)
+option(TF_PRE_BUILT_LOCATION "Location where TensorFlow CC library is located" FALSE)
+option(NGRAPH_DISTRIBUTED_ENABLE "Add distributed mode to the CPU backend" FALSE)
+option(NGRAPH_PLAIDML_ENABLE "Build PlaidML backend" FALSE)
+
+# Validat the options
 if (NGRAPH_PLAIDML_ENABLE)
     if (NOT DEFINED PlaidML_DIR)
         message(FATAL_ERROR "Provide the PlaidML SDK directory: -DPlaidML_DIR=<directory>")
     endif()
     message(STATUS "nGraph/PlaidML enabled")
 endif()
 
-# Find and build ngraph
-option(NGRAPH_DISTRIBUTED_ENABLE "Add distributed mode to the CPU backend" FALSE)
-if(NGRAPH_DISTRIBUTED_ENABLE)
-    find_package(MPI REQUIRED)
-    add_definitions(-DNGRAPH_DISTRIBUTED)
-    include_directories(SYSTEM ${MPI_C_INCLUDE_PATH} ${MPI_CXX_INCLUDE_PATH})
-    link_directories(${MPI_C_LIBRARIES} ${MPI_CXX_LIBRARIES})
+if(USE_PRE_BUILT_NGRAPH)
+    if(NOT NGRAPH_ARTIFACTS_DIR)
+        message(
+            FATAL_ERROR 
+            "USE_PRE_BUILT_NGRAPH is ON but NGRAPH_ARTIFACTS_DIR is missing"
+        )
+    endif()
+    # Create absolute path for the directory
+    get_filename_component(
+        NGRAPH_ARTIFACTS_DIR
+        "${NGRAPH_ARTIFACTS_DIR}"
+        ABSOLUTE
+    )
+    if (NOT EXISTS ${NGRAPH_ARTIFACTS_DIR})
+        message(FATAL_ERROR 
+            "nGraph artifacts directory doesn't exist: " ${NGRAPH_ARTIFACTS_DIR} )
+    endif()    
```

Comment:
Is it possible to use only `NGRAPH_ARTIFACTS_DIR` and skip the `USE_PRE_BUILT_NGRAPH` flag completely, since the presence of one implies the other.

Some thing like:
```
if(NGRAPH_ARTIFACTS_DIR)
     # Create absolute path for the directory
     get_filename_component(
         NGRAPH_ARTIFACTS_DIR
         "${NGRAPH_ARTIFACTS_DIR}"
         ABSOLUTE
     )
     if (NOT EXISTS ${NGRAPH_ARTIFACTS_DIR})
         message(FATAL_ERROR 
             "nGraph artifacts directory doesn't exist: " ${NGRAPH_ARTIFACTS_DIR} )
     endif()    
endif()
```